src/components/Post/Post.js

```diff
@@ -0,0 +1,29 @@
+import React, {Component} from 'react';
+import Checkbox from '@material-ui/core/Checkbox';
+import classes from './Post.module.css';
+
+class Post extends Component {
+    state = {
+        checked: false
+    }
+
+    handleChange = () => {
+        this.setState({checked: !this.state.checked});
```

Comment:
It is usually not preferred to override existing state directly as following as the setState is an async operation 
Please change to (prev) => ({checked: !prev.checked })